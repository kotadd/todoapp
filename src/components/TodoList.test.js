import React from 'react';
import { shallow } from 'enzyme';

import TodoList from './TodoList';

describe('TodoList component', () => {
  let wrapper;
  let mockTodos;
  let mockToggleTodo;

  beforeEach(() => {
    mockTodos = [
      {
        id: 1,
        text: 'mock todo',
        completed: false
      }
    ];

    mockToggleTodo = jest.fn();

    const mockProps = {
      todos: mockTodos,
      toggleTodo: mockToggleTodo
    };
    wrapper = shallow(<TodoList {...mockProps} />);
  });
  it('should render TodoList component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleTodo with the right id when Todo clicked', () => {
    wrapper.find('Todo').simulate('click');

    expect(mockToggleTodo).toHaveBeenCalled();
  });
});
