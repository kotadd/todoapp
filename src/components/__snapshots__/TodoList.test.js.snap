// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should render TodoList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    todos={
      Array [
        Object {
          "completed": false,
          "id": 1,
          "text": "mock todo",
        },
      ]
    }
    toggleTodo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Todo
          completed={false}
          id={1}
          onClick={[Function]}
          text="mock todo"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "completed": false,
          "id": 1,
          "onClick": [Function],
          "text": "mock todo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Todo
            completed={false}
            id={1}
            onClick={[Function]}
            text="mock todo"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "completed": false,
            "id": 1,
            "onClick": [Function],
            "text": "mock todo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
